### DISCUSSION TITLE
Can't create a new Uno 5.0 default app

### DISCUSSION THREAD

**Discussion by MarkBrainfire** [unoplatform/uno#14268] [General]
I was excited to try out the new Uno 5.0.  No luck in VS.
The new templates don't show up at all in VS Prelease.
In the Visual Studio released version 17.7.6 they show up but don't work.
First you get a message about needing to close the build and reload projects.
Then the dependencies complain that you rolled back WindowsAppSDK but when you attempt to update this it is "blocked by project".

The default app should just work.  I'm not an edge case.  I am using Visual Studio.  I've run Uno-Check a million times.

So, I ought to be able to create the default app without any problems.

How are things tested before a release?  The process needs to be improved.

I'm tired ... I'll come back and provide more details later if this is helpful ...

**Comment by MarkBrainfire** [comment]
No one has commented in this discussion???  Am I the only one in the world having difficulty creating Uno Platform 5.0 applications?  If you are also having problems SPEAK UP!!!

The entire process of creating a new app in Visual Studio is completely flakey.  First the new templates do not appear in the preview version of Visual Studio, so you have to create a new app in the released version of Visual Studio.  But once created (if you get that far) the default solution will NOT run in the released version of Visual Studio so you have to run it in the Preview Version.

To try and sort out the problems I created 12 different new default solutions, 6 variations repeated twice.  I'll share what I found here in the hopes that this information will help whoever is trying to make Uno actually work.

The released version of VS I'm using is 17.7.6 and the Preview version is 17.8.0 Preview 6.0.
For all 12 attempts to create a new default Uno App the following were the same:

- Created using VS 17.7.6, run in VS Preview 17.8.0 Preview 6.0
- .NET 7
- Created via the Wizard by customizing the Blank Application
- Target platforms: Windows, Android
- Presentation: None
- Theme: Fluent
- Extensions: Blank, Console

The anomalies experienced are:
- Almost every time when the welcome message appears, there is also an error message that says: "The build must be stopped before the project can be closed."
- EVERY time Visual Studio displays a brown bar message at the top that requires me to reload the projects.
- Most of the time the projects are NOT properly nested under the Solution Folders so that the Source and Platforms folders are empty and all projects are displayed under the Solution root.
- EVERY time the solution fails to compile and deploy in VS 17.7.6 giving the error message that wasm-tools is not present.
- Some of the time (when both Markup and Maui are selected) an addition versioning error re: WindowsAppSDK also appears.
- When the WindowsAppSDK versioning error appears the solution will not compile using VS Preview 17.8.0 Preview 6.0, otherwise it will.  

These anomalies for my key test cases are listed below:

CPM = Central Package Management, Maui = Maui embedding, Folders = Solution Folder Anomaly, 
wasm  = wasm-tools error, WASDK = WindowsAppSDK versioning error.

UI          /   Features        /    Folders   /     wasm   /     WASDK  /     Runs in Preview
XAML    /   CPM, Maui    /    Yes      /       Yes     /        No      /          Yes
Markup /   CPM, Maui   /     No     /        Yes    /         Yes      /           No
Markup /  none             /     Yes     /       Yes     /        No       /           Yes
Markup /   Maui            /     Yes    /        Yes     /        Yes       /           No
Markup /   CPM             /     No   /         Yes    /         No       /           Yes

From this table it can be seen that Markup and Maui embedding are incompatible due to the WindowsAppSDK versioning error.  Attempts to update the WindowsAppSDK fail saying the project is blocking the update.

Uno Platform 5.0 looks promising when you finally get it working with Visual Studio.
I look forward to that day.

Mark

**Reply by MartinZikmund** [reply]
Would you be willing to record yourself performing the steps above or connect with us and show us these problems on a call? It would help us investigate this and see the problems through your eyes.

**Reply by dansiegel** [reply]
> No one has commented in this discussion???

@MarkBrainfire please keep in mind you opened this discussion on a Friday night. I'm just waking up to this and looking into your issue as soon as I got up, so I can assure you that your issue isn't being ignored.

You've brought up a few scenarios that I will need to look into... As for the 'Project must be reloaded' issue, that should be fixed in the next update to the VS Extension.

**Comment by kazo0** [comment]
Thanks for the report @MarkBrainfire ! 

Sorry that you are experiencing issues, hopefully we can get these solved for you ASAP

> EVERY time Visual Studio displays a brown bar message at the top that requires me to reload the projects.

![image](https://github.com/unoplatform/uno/assets/4793020/1997fb96-5e8a-4e62-9c71-491575e05396)

This message is expected and will disappear once you select "Reload Projects". You can see it mentioned in the [Wizard - Getting Started tutorials](https://platform.uno/docs/articles/getting-started/wizard/startup.html#customization). I'm not quite sure of the reason it appears, I'm assuming a limitation with Visual Studio for third-party extensions, perhaps @jeromelaban knows?

> EVERY time the solution fails to compile and deploy in VS 17.7.6 giving the error message that wasm-tools is not present.

`wasm-tools` is a dotnet workload that is connected to the dotnet SDK version that is being used to build the project. Depending on whether you are using VS Stable or VS Preview, can you try running `uno-check` or `uno-check --pre`? Refer to the [uno-check documentation](https://platform.uno/docs/articles/external/uno.check/doc/using-uno-check.html). 

If all else fails, perhaps try and run `dotnet workload install wasm-tools`. 

If nothing else works, can you navigate to the root folder of the failing solution and run `dotnet --info` and paste the output here?

> Some of the time (when both Markup and Maui are selected) an addition versioning error re: WindowsAppSDK also appears.
> When the WindowsAppSDK versioning error appears the solution will not compile using VS Preview 17.8.0 Preview 6.0, otherwise it will.

This one I am able to reproduce when using Markup and MAUI Embedding and the appropriate team members have been notified and will look into it ASAP, thanks for finding this!
 
For the remaining issues:

> Almost every time when the welcome message appears, there is also an error message that says: "The build must be stopped before the project can be closed."

I can't get this one to repro so we will need to keep investigating

> Most of the time the projects are NOT properly nested under the Solution Folders so that the Source and Platforms folders are empty and all projects are displayed under the Solution root.

This one I can also not reproduce but is definitely concerning, this one would be good to get a recording of like @MartinZikmund mentioned so we can narrow it down and fix ASAP


Let me know if you get more info or if anything here helps, let's get these fixed for you :)

**Reply by MarkBrainfire** [reply]
Hi Steve
,  
Re: wasm-tools, nothing works.  See this: https://github.com/unoplatform/uno/discussions/14191

Of course, I've run uno-check and uno-check --pre-major multiple times.  Sometimes it appears to require fixing something and then even after a reboot when run again it wants to fix it again. 

Yes, I'll make a vid of attempting to create a new app.

Thanks for your help.

Mark

**Comment by MarkBrainfire** [comment]
@MartinZikmund @kazo0 @dansiegel 

Martin requested videos ...

Here's my uno-check:
https://iframe.mediadelivery.net/play/170397/888a256c-f957-430d-a3b7-fe01c32bd814

Here's my uno-check --pre-major:
https://iframe.mediadelivery.net/play/170397/501bf76b-a513-47fb-a105-ae6d3755e690

Here's an attempt to create an Uno Platform 5.0 XAML app:
https://iframe.mediadelivery.net/play/170397/13bf1031-e5c6-4694-bf34-e09d0da56d38

Here's an attempt to create an Uno Platform 5.0 Markup / Maui Embedding app:
https://iframe.mediadelivery.net/play/170397/67cfa5b6-929f-4eb2-9b20-db2043d6839c

**Reply by jeromelaban** [reply]
Thanks for the videos.

Multiple pieces are at play here. First, you're using preview builds of .NET and Visual Studio, which may cause issues that we may not have encountered yet, particularly when .NET 7 and .NET 8 are used simultaneously.

- For the missing workload, this is unfortunately a best effort on Uno's part. Workloads are a difficult part of the .NET framework, and while it's improving with .NET 8, there are only two known workarounds:
  - Run the new .NET 8 clean command:

     ```bash
     dotnet workload clean -all
     ```
     which will remove all workloads then allow for `uno-check` to properly reinstall them
  - If it still does not work, reinstall visual studio, which will reinstall .NET entirely but as you have multiple versions of VS installed, you may need to reinstall both.
- For the `build cannot be stopped` error, similar to @kazo0, we'll be looking into it, are we're not able to reproduce at this time.
- For the errors you've noticed with C# markup and the incorrect versions of WinAppSDK, it looks like the template being applied is an incorrect one. Can you try uninstalling the dotnet new templates using `dotnet new uninstall uno.templates`, then try to create a project again ? (Don't try to run uno-check right after as it tries to reinstall the templates, and we'll see if it has an impact).
- Try creating apps using the [`dotnet new unoapp`](https://platform.uno/docs/articles/get-started-dotnet-new.html?tabs=net7%2Cwindows) template, and see if the errors are the same.

**Reply by MarkBrainfire** [reply]
Thanks Jerome,

Here's a report on what I've found so far ...
dotnet -workload clean --all  didn't seem to do much and didn't solve any problems.
Uninstalling both versions of Visual Studio and reinstalling just the released version (17.7.6) DID solve two problems:
1) The pesky "build cannot be stopped" messages went away
2) The solution was organized in the proper solution folders

One interesting thing .. even after reinstalling Visual Studio and redundantly also reinstalling the .NET SDK (403), dotnet commands insisted on looking for the dotnet.dll in the .net rc 2 folder (c:\Program Files\dotnet\sdk\8.0,100-rc.2.23502.2) even though the dotnet.dll was in c:\Program Files\dotnet\sdk\7.0.403.  I didn't know how to redirect it.  Even setting an Environment path to the correct folder didn't work.  So I gave in and just re-installed .NET 8 rc 2, which of course populated the c:\Program Files\dotnet\sdk\8.0,100-rc.2.23502.2 folder with all of the SDK files, including dotnet.dll.  

I've noticed that uno-check insists on installing only 7.0.402 of the SDK.  It downloads this version of the SDK but then doesn't install anything when it detects a more recent version is already installed.  This is good functionality, but it seems a bit clumsy to download it and then twice abort installation when the later version (403) is detected.

The inability to run the default Uno Platform 5.0 app in VS remains when any mobile target platform is selected due to VS not finding wasm-tools.  This problem persists even after installing wasm-tools by going to the project directory and running the installation which reports that it has been successfully installed.  As I noted in my prior discussion thread, this can be "solved" by running the solution in the preview version of VS, which is able to find the wasm-tools and successfully run the solution.  In my case I haven't re-installed the preview version of VS yet, because I don't want to complicate the variables involved until  I can get things to work with the released 17.7.6 version.

When Markup and Maui are selected the WindowsAppSDK versioning problem appears.  In this regard I've noticed that the versioning of both WindowsAppSDK and Windows.Build.Tools are handled inconsistently depending on whether in the Uno configuration wizard you begin with the blank (left) starting point or the default (right)  starting point -- even if you end up selecting nearly the same set of options ... and I suspect this is the problem. When you start with the default case, the latest WindowsAppSDK version (1.4) is selected, however when you start with the blank case, one of the packages uses WindowsAppSDK 1.2 and this leads to the error message and prevents compiling.  

I'll continue to play around with things to see if I can get the default version of Uno 5.0 to run on the released version of VS before installing the VS preview again.

**Reply by jeromelaban** [reply]
Thanks for the update. It's generally a good idea to provide logs for operations that don't work (`dotnet workload clean --all` would have been interesting to look at), we'll generally ask for anything that can help troubleshoot further.

A few points:
1. Can you provide the details about the last two points I mentioned above?
2. Also, can you zip and attach here the folder of the app that was created which is not compiling properly with the winappsdk error?
3. For the `wasm-tools` not working, it likely still means that the dotnet workloads are not installed properly, and they'll keep failing until this is somehow returning on its feet. @kazo0 asked for a log of `dotnet workload install wasm-tools`, can you post it here as well?

As for the clumsiness of the tooling, we're always accepting contributions in [uno-check](https://github.com/unoplatform/Uno.check) if you know how to improve the process.

Thanks.

**Reply by MarkBrainfire** [reply]
re: "we're always accepting contributions .."
I doubt I'm qualified to contribute much.  Once I'm up to speed with MSBuild and the intricacies of Uno Platform, I'll be happy to contribute.  So many moving parts ... I'm in awe with what you and your team have put together!

Yes I did re-install the new templates.  I didn't try dotnet new unoapp.  So I'll revisit both of these suggestions now.

Yes, I'll capture and forward logs for wasm-tools and workload clean once I figure out how to do this.

**Comment by MarkBrainfire** [comment]
@jeromelaban @kazo0 

Here is the log file for dotnet workload clean --all
[clean-all-log.txt](https://github.com/unoplatform/uno/files/13260806/clean-all-log.txt)

Here is the log file installing wasm-tools from the projects dir of an attempted new Uno 5.0 app
[wasm-log.txt](https://github.com/unoplatform/uno/files/13260815/wasm-log.txt)

I tried running dotnet new unoapp -n cmdlineapp --maui-embedding -markup csharp
I got many errors about workloads not being installed.  
I went to the solution directory and used dotnet workload restore to install the missing workloads.
Then, inside visual studio I got a brown bar message (within the solution explorer only) that I needed to install things for the solution to compile.  I did so.

When I attempted to compile, I still got the error about wasm-tools workload not being present and the WindowsAppSDK  versioning error.

Mark

**Comment by MarkBrainfire** [comment]
C:\Users\Mark>dotnet new uninstall uno.templates
Success: Uno.Templates::5.0.13 was uninstalled.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.cs.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.de.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.en.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.es.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.fr.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.it.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.ja.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.ko.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.pl.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.pt-BR.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.ru.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.tr.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.zh-Hans.json)", it will be skipped from further processing.
Warning: Failed to read or parse localization file "C:\Program Files\dotnet\templates\8.0.0-rc.2.23480.2\microsoft.dotnet.web.projecttemplates.8.0.8.0.0-rc.2.23480.2.nupkg(/content/BlazorWeb-CSharp/.template.config/localize/templatestrings.zh-Hant.json)", it will be skipped from further processing.

**Comment by MarkBrainfire** [comment]
When I go into VS and manage extensions Uno Platform project templates are shown as installed.

**Comment by MarkBrainfire** [comment]
I just installed VS Preview (17.8 Preview 6) and confirmed that it will successfully run apps that would not run in VS 17.7.6 due to the wasm-tools error.  In other words, the preview version seems to solve the wasm-tools issue -- without needing to install any additional workloads.

I also confirmed that projects created using VS 17.7.6 incorporating MAUI and Markup have the same WinAppSDK error:

![WinAppSDKError](https://github.com/unoplatform/uno/assets/78389339/b2fac41c-ec84-41a6-bb3f-8a83456d4b23)

Using VS Preview I used "manage extensions" to install Uno Templates. After doing so, I am able to create a "plain" (no Maui or Markup) XAML app which compiles and runs just fine.

Using VS Preview I am also able to create a Markup app (Window / Android, Central Package) that compiles and runs just fine.

In both these successful cases I'm using the default wizard customized for  .NET 7, only Windows / Android, MVUX, Fluent, Blank / Console, and no testing.

However, even using VS Preview I am not able to compile an app incorporating both Markup and Maui Embedding due to the same WinAppSDK versioning error:
![SameWinAppSDKErrorInPre](https://github.com/unoplatform/uno/assets/78389339/38d2ad23-3b67-4ec5-b803-8fc04d7fabaf)

**Comment by nickrandolph** [comment]
Hi @MarkBrainfire 

The issue with WinAppSdk version can be resolved by adding the ItemsGroup to the "When Condition="$(IsWinAppSdk)" in the main class library csproj (in your case the UnoMauiMarkupPre.csproj). I'm updating the template as we speak to avoid this issue in the future

```xml
    <When Condition="$(IsWinAppSdk)">
      <ItemGroup>
        <PackageReference Include="Microsoft.WindowsAppSDK" VersionOverride="1.4.231008000" />
        <PackageReference Include="Microsoft.Windows.SDK.BuildTools" VersionOverride="10.0.22621.756" />
      </ItemGroup>
    </When>
```

**Reply by nickrandolph** [reply]
see issue raised for this: https://github.com/unoplatform/uno.templates/issues/396

**Reply by MarkBrainfire** [reply]
Thanks Nick @nickrandolph ,
I just uninstalled and reinstalled the templates (ver: 5.0.9.11) in both the released VS 17.7.6 and the preview version 17.8 P6
They don't seem to add the conditional ItemGroup you listed above when you select the options (Markup, Maui, CPM) I'm using.  The WinAppSDK versioning error persists.   

I'll keep experimenting and attempt to modify the project files manually to see if I can overcome the WinAppSDK version issue using VS Pre.  I guess no one has yet resolved the wasm-tools issue for VS 17.7.6.  That's ok, if we can get Uno Platform 5.0 to build in the preview version of VS I am happy with that.

In the app I'm developing, I'm making extensive use of the MAUI embedding feature, so it's important for me to get this working.  I'm excited to experiment with switching from XAML to Markup since it appears to offer many advantages (compile time validation and the ability to call methods etc.).  So, if we can get all of this working it would be excellent!

**Reply by jeromelaban** [reply]
> They don't seem to add the conditional ItemGroup you listed above when you select the options (Markup, Maui, CPM) I'm using. The WinAppSDK versioning error persists.

This is not released yet, the suggestion was for you to add this block in your solution to see if it helps.

**Reply by MarkBrainfire** [reply]
Ok, I'll try it.  I'm just learning about MSBuild and Central Package Management.  Enabling Central Package Management lists the desired PackageVersions as a single ItemGroup in the Solution Items Directory.Packages.props file, is that right?  Should the WinAppSDK and SDK.BuildTools versions Nick listed above be added to that file as well?  

I assume I'll be adding the conditional When group to the common Class Library.  Do I also need to add it to the Windows head project?

**Comment by MarkBrainfire** [comment]
@jeromelaban @nickrandolph 

Using VS Preview
To keep things simple: no central package management.  Windows only.  With Markup and Maui.

Nick's version overrides (in the common class library) make the dependencies happy, BUT then I'm prevented from successfully compiling with this error:
![Manifest Error](https://github.com/unoplatform/uno/assets/78389339/aa73f199-1a6a-4fdd-8a10-c535aea7597c)

**Reply by jeromelaban** [reply]
Thanks can you post a [binlog of the build](https://platform.uno/docs/articles/uno-builds-troubleshooting.html) please? Make sure to do so for every build error you report. Screenshots are difficult to manage and cannot be searched easily. Thanks!

**Reply by jeromelaban** [reply]
@MarkBrainfire it worked, I can get the binlog. Thanks. To make the threads more easy to read, make sure to respond to the same thread (Yes, the GitHub UX for this is not the most intuitive...)

**Reply by jeromelaban** [reply]
Thanks, so at this point, it's best for you to wait for the updated template that should come out in a few hours, if you don't want to fiddle further with csproj files.

Still, what you're getting here is that the WinAppSDK package that is used by the project is `1.0.0` (the path in the error is `C:\Users\Mark\.nuget\packages\microsoft.windowsappsdk\1.0.0\build` and should be `1.4.231008000`), which is not supported anymore (and had many issues).

NuGet seems to have restored an incorrect version in your build, most likely caused by an earlier incorrect version specified in the csproj. You can try deleting your `bin` and `obj` folder, then build again so the packages get restored properly.

**Comment by MarkBrainfire** [comment]
Here's a binlog uploaded to dropbox.  Let me know if you are not able to view it.  Thanks.

https://www.dropbox.com/scl/fi/6gmn37f8h3bb0uakdz341/UnoX.binlog?rlkey=wxutigqz7if208lschmhmtc8z&dl=0

**Comment by MarkBrainfire** [comment]
After updating to the latest versions of uno-check and the latest Uno Platform templates for VS Preview, I was able to create and run an app with both Markup and MAUI embedding.  THANK YOU!  So, I can now proceed with my Brainfire project.  

However, for some of the other variations there were still problems.  Perhaps one of the best ways I can contribute to Uno Platform is performing and documenting tedious testing.  For now, my way to give back.

Tomorrow I'll post a detailed report of many Uno app combinations.  For any combination with problems, I'll include a binlog.  

I just updated VS to Preview 17.8.0 Preview 7.  Perhaps this will make a difference as well.

The hard work of the Uno Platform team is MUCH appreciated by me.  

Mark

**Reply by jeromelaban** [reply]
Thanks for the update! Let us know which issues you find, it may be that we have not found all of them with all combinations.

**Reply by MarkBrainfire** [reply]
As you know the wasm-tools issue prevents compiling Uno solutions in the released version of VS (17.7.6)

Since just upgrading to preview 7 of 17.8 I can't create solutions at all. It just creates solution items without projects. 

I'll try to revert to preview 6 or 5.

**Reply by jeromelaban** [reply]
Thanks for the update. When running the wizard finishes to start, there's a `uno platform` output window that generally gives some information, can you paste the contents here?

As for the `wasm-tools`, we do not have a repro and this makes it difficult for us to troubleshoot or give details to Microsoft. Still could you try running a `dotnet workload restore -v diagnostic` or `dotnet workload install -v diag wasm-tools` ? I have not found, in all the discussions we've had so far, seen a full error log of that command that you ran.

**Reply by MarkBrainfire** [reply]
Hi Jerome,

To debug the wasm-tools issue, which no one else seems to be experiencing, I've decided to learn MSBuild and MSBuild debugging.  To unravel this issue, I'll be using logs of dotnet commands, binarylogging from MSBuild, and MSBuild StructuredLogViewer.

Are there any other tools I should consider for debugging builds in Visual Studio?

Mark

**Reply by jeromelaban** [reply]
Before you do that, can you provide the full logs for the error for the commands I provided? The issue you're getting is related to workloads installation, not necessarily msbuild.

**Reply by MarkBrainfire** [reply]
I'm unable to run dotnet workload restore while in the common libary project directory.
Here's what happens (text file via Bunny CDN):
https://issues.b-cdn.net/ErrorsAttemptingToRun%20dotnet%20workload%20restore.txt

I'm also unable to run dotnet workload install wasm-tools.  
Here's what happens (2nd text file via Bunny CDN):
https://issues.b-cdn.net/ErrorsAttemptingToRun%20dotnet%20workload%20install%20wasm-tools.txt

I was able to run both dotnet commands previously, but it didn't help.  However, since then I've uninstalled and reinstalled VS and VS Preview.  So, obviously I've screwed something up.  

This "UnoWAX" project (WAX = Windows, Android, XAML) won't compile due to the absence of wasm-tools.  However, I created a "UnoMin" project (just Windows and XAML) and it compiled and ran fine.  uno-check doesn't report any problems.  Previously, I was able to "solve" the wasm-tools issue by using the Preview version (17.8 Preview 6) of VS.  Then I subsequently installed the update to Preview 17.8 Preview 7 and this no longer worked at for uno at all... not creating any projects at all, but only a Solution Items folder.

**Reply by jeromelaban** [reply]
Thanks for the logs. 

It really feels like .NET is corrupted beyond the repair procedures. I would suggest uninstalling all visual studio instances, then remove the dotnet folder in "program files" completely and start over.

The problem with repair tools is that they can't remove pieces that have been added by visual studio, and it seems that repairing VS instances individually does not help.

**Reply by MarkBrainfire** [reply]
@jeromelaban After installing the newly released VS 17.8, ... everything just started working.
Looks like I can get back to coding my application.  Thank God! (and you and your team).

I haven't tried Uno Platform with .NET 8 yet.  I'll wait until all references to .NET 8 rc2 are replaced.

Thanks, and congratulations.
Mark

**Reply by jeromelaban** [reply]
Thanks for the update @MarkBrainfire! uno-check updates for .NET 8.0.100 will be out today, as well as other changes in the templates.

**Comment by plppp2001** [comment]
Hi All,

Today I'm receiving the following error, not sure what's up with this;
![image](https://github.com/unoplatform/uno/assets/45370058/2261b21e-b27d-4e82-92b1-262caa18be27)

**Reply by jeromelaban** [reply]
This looks like some fluke of the .NET workloads. You can try running the mentioned command.

**Reply by plppp2001** [reply]
I read somewhere else to run the following, and it's currently running (taking a while):
"dotnet workload repair"
![image](https://github.com/unoplatform/uno/assets/45370058/12a3293d-a635-49dc-869e-523a8765b692)

**Reply by plppp2001** [reply]
VS2022 - Version 17.10.1

@jeromelaban  After running "dotnet workload repair" and then restarting my WIndows 11 comp. The issue went away.  (I think this error popped up maybe.... because I started installing and messing around with the new Aspire, some preview version... not sure)

thanks.

**Comment by bdrajer** [comment]
RE "The build must be stopped before the project can be closed"

Every time I change the debug target (for example, from Desktop to WebAssembly) using the button in the top toolbox, the VS project seems to get reloaded. If I immediately run the app, I get the mentioned error. If I wait for a couple of seconds, no error appears. In the second case I see that the run/debug button gets disabled for a moment, supposedly signalling that I cannot run the app while the project is being reloaded, but this happens with a big delay (around 5 seconds after I change the debug target), so there's plenty of time for me to click the button and trigger the error.

**Reply by MartinZikmund** [reply]
There are several issues at play here, but unfortunately all out of our control: 

https://github.com/unoplatform/uno/issues/16516#issuecomment-2085032308
